File: java_pro/src/main/java/jv/com/tencent/cain/design_pattern/state/ResetState.java
package jv.com.tencent.cain.design_pattern.state;

/**
 * @author cainjiang
 * @date 2020/3/27
 */
public class ResetState extends State {
    @Override
    public void doWork(Work work) {
        System.out.println("当前时间：" + work.getHour() + ",终于可以下班休息了");
    }
}

File: java_pro/src/main/kotlin/kt/com/tencent/cain/desigin_pattern/state/ResetState.kt
package kt.com.tencent.cain.desigin_pattern.state

/**
 * @author : jiangyu
 * @date   : 2020/3/28
 * @desc   : xxx
 */
class ResetState : State() {
    override fun doWork(work: Work) {
        println("下班了")
    }
}